# KNeighborsClassifier
-------
- KNN without train test split (not used any where in live)

## Step 1: Understanding Data
-----------

In [0]:
# Import KNeighborsClassifier from sklearn.neighbors
from sklearn.neighbors import KNeighborsClassifier
from sklearn import datasets
import numpy as np

In [0]:
KNeighborsClassifier?

In [0]:
iris = datasets.load_iris()

In [0]:
X=iris["data"]

In [0]:
y=iris["target"]

In [0]:
type(X)

numpy.ndarray

In [0]:
X.ndim

2

In [0]:
X.shape

(150, 4)

In [0]:
X[:5]

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2]])

In [0]:
type(y)

numpy.ndarray

In [0]:
y.ndim

1

In [0]:
y.shape

(150,)

In [0]:
y[:5]

array([0, 0, 0, 0, 0])

## Step 2: Fit The model
-------------

In [0]:
knn = KNeighborsClassifier(n_neighbors=6) # how we will know 6 ????
knn

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
           metric_params=None, n_jobs=None, n_neighbors=6, p=2,
           weights='uniform')

In [0]:
f = knn.fit(X, y)
f

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
           metric_params=None, n_jobs=None, n_neighbors=6, p=2,
           weights='uniform')

## Step 3: Predict
-------------
- Now you are walking in the flower garden of setosa ,versicolor and verginica
- found three flowers and measured length/width and asking machine to find the labels

In [0]:
x_new = np.array([[5.1,3.5,1.4,0.2], [5.5,2.3,4.,1.3],[6.7,3.3,5.7,2.5]], dtype=np.float)
prediction = knn.predict(x_new)
print(prediction)

[0 1 2]


- Note: In this we used all known 150 samples/rows used in traing the model
- can you comment the machine answers are corect or wrong

    <input type="radio" disabled Checked> Can not Comment
    
    <input type="radio" disabled> can Comment

## Step 4: Measuring model performance
---------------
1. In classification, accuracy is a commonly used metric
2. Accuracy = Fraction of correct predictions


- As we are not in a positions to justify the machine answers are right /wrong . So we can not calculate performance/accuracy